src/libsync/propagatorjobs.cpp

```diff
@@ -54,6 +55,10 @@ QByteArray localFileIdFromFullId(const QByteArray &id)
  */
 bool PropagateLocalRemove::removeRecursively(const QString &path)
 {
+#ifdef Q_OS_UNIX
+    _moveToTrash = propagator()->syncOptions()._moveFilesToTrash;
```

Comment:
Since removeToTrash is already recursive, you should not call this function when moving to trash.
It is already complicated enough.  But all the complexity comes from the fact we might have individual faillure for individual files, which can't happen here.

Don't use this code path when moving to trash.